File: validAnagram.js
class Solution {
  /**
   * @param {string} s
   * @param {string} t
   * @return {boolean}
   */
  isAnagram(s, t) {
    if (s.length !== t.length) {
      return false;
    }

    let sHash = {},
      tHash = {};

    for (const sChar of s) {
      sHash[sChar] = 1 + (sHash[sChar] || 0);
    }

    for (const tChar of t) {
      tHash[tChar] = 1 + (tHash[tChar] || 0);
    }

    for (const key in sHash) {
      if (sHash[key] !== tHash[key]) {
        return false;
      }
    }
    return true;
  }
}

File: valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_hash, t_hash = {}, {}

        for s_char in s:
            s_hash[s_char] = 1 + s_hash.get(s_char, 0)

        for t_char in t:
            t_hash[t_char] = 1 + t_hash.get(t_char, 0)

        return s_hash == t_hash
